sctp/src/association/association_internal.rs

```diff
@@ -553,62 +515,42 @@ impl AssociationInternal {
                 cumulative_tsn_ack: self.cumulative_tsn_ack_point,
             };
 
-            if let Ok(raw) = self.create_packet(vec![Box::new(shutdown)]).marshal() {
-                if let Some(t2shutdown) = &self.t2shutdown {
-                    t2shutdown.start(self.rto_mgr.get_rto()).await;
-                }
-                raw_packets.push(raw);
-            } else {
-                log::warn!("[{}] failed to serialize a Shutdown packet", self.name);
+            let p = self.create_packet(vec![Box::new(shutdown)]);
+            if let Some(t2shutdown) = &self.t2shutdown {
+                t2shutdown.start(self.rto_mgr.get_rto()).await;
             }
+            raw_packets.push(p);
         } else if self.will_send_shutdown_ack {
             self.will_send_shutdown_ack = false;
 
             let shutdown_ack = ChunkShutdownAck {};
 
-            if let Ok(raw) = self.create_packet(vec![Box::new(shutdown_ack)]).marshal() {
-                if let Some(t2shutdown) = &self.t2shutdown {
-                    t2shutdown.start(self.rto_mgr.get_rto()).await;
-                }
-                raw_packets.push(raw);
-            } else {
-                log::warn!("[{}] failed to serialize a ShutdownAck packet", self.name);
+            let p = self.create_packet(vec![Box::new(shutdown_ack)]);
+            if let Some(t2shutdown) = &self.t2shutdown {
+                t2shutdown.start(self.rto_mgr.get_rto()).await;
             }
+            raw_packets.push(p);
         } else if self.will_send_shutdown_complete {
             self.will_send_shutdown_complete = false;
 
             let shutdown_complete = ChunkShutdownComplete {};
+            ok = false;
+            let p = self.create_packet(vec![Box::new(shutdown_complete)]);
 
-            if let Ok(raw) = self
-                .create_packet(vec![Box::new(shutdown_complete)])
-                .marshal()
-            {
-                raw_packets.push(raw);
-                ok = false;
-            } else {
-                log::warn!(
-                    "[{}] failed to serialize a ShutdownComplete packet",
-                    self.name
-                );
-            }
+            raw_packets.push(p);
         }
 
         (raw_packets, ok)
     }
 
     /// gather_outbound gathers outgoing packets. The returned bool value set to
     /// false means the association should be closed down after the final send.
-    pub(crate) async fn gather_outbound(&mut self) -> (Vec<Bytes>, bool) {
-        let mut raw_packets = vec![];
+    pub(crate) async fn gather_outbound(&mut self) -> (Vec<Packet>, bool) {
+        let mut raw_packets = Vec::with_capacity(16);
```

Comment:
Seemed like a good middle-ground value that would fit most common scenarios I saw without wasting too much memory. At it saves a lot of reallocations because rust vecs go in steps of power of two. So collecting 3 packets would mean 3 reallocations.